packages/trpc/server/routers/viewer/teams/inviteMember/utils.ts

```diff
@@ -178,14 +178,10 @@ export async function findUsersWithInviteStatus({
   const inviteesFromDb: UserWithMembership[] = await prisma.user.findMany({
     where: {
       OR: [
-        // Either it's a username in that organization
+        // Either it's a username with organization as null in the main user table.
         {
-          profiles: {
-            some: {
-              organizationId: team.id,
-              username: { in: usernamesOrEmails },
-            },
-          },
+          username: { in: usernamesOrEmails },
+          organizationId: null,
```

Comment:
Ah, this is not the correct way. You can no longer invite users using usernames in a team within the organization. Your code is almost correct, but the catch is that instead of setting the organizationId to null, we need to do it like this :- 
```
const orgInfo = team.isOrganization && {
  organizationId: team.id,
}
```
and then 

```
{ username: { in: usernamesOrEmails }, ...orgInfo }, { email: { in: usernamesOrEmails } }

```

And yes, usernames are also unique. However, our job isn't finished yet. There might be functions or hooks that only allow email invitations and don't account for usernames, so you might encounter some undefined errors.

Here are my thoughts on this. Please let me know what you think and try to solve it using my suggestion. If you have any other methods to solve this, feel free to share.

Reply:
@anikdhabal

> Still, you cannot invite users using usernames in a team within the organization.

Yup correct!

I have made some different change instead of the one you mentioned above.

The code you gave particularly this one -

```
const orgInfo = team.isOrganization && {
    organizationId: team.id,
  };
```
This one seems to address when you are adding an user to "an organization" not a "team of an organization".
And while I agree that currently you can't add an user to an organization, but that's disabled in front-end also -

<img width="1511" alt="Screenshot 2024-07-23 at 3 08 39 PM" src="https://github.com/user-attachments/assets/6c4a6674-e274-4636-9876-aa2b6167ff6e">

That's why I think we can handle that case later, in a different PR. What do you think?

With my last commit I have added some changes to support adding users to team or an organization also.

cc: @hariombalhara